### PR TITLE
feat(discover): snql support for no arg functions

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Implemented snql support for count, last_seen, latest_event, and failure_rate.
Added tests for the above functions.

### CODE REVIEW

File: src/sentry/search/events/fields.py [hunk 1]
```diff
@@ -2178,6 +2178,42 @@ def __init__(self, dataset: Dataset, params: ParamsType):
...
+                    default_result_type="integer",
+                ),
+                SnQLFunction(
+                    "last_seen",
+                    snql_aggregate=lambda _, alias: Function(
+                        "max",
+                        [self.column("timestamp")],
+                        alias,
+                    ),
+                    default_result_type="integer",
```
Comment (shruthilayaj) [inline src/sentry/search/events/fields.py:2197]: does this need to be date type?
```suggestion
                    default_result_type="date",
```

File: src/sentry/search/events/fields.py [hunk 2]
```diff
@@ -2178,6 +2178,42 @@ def __init__(self, dataset: Dataset, params: ParamsType):
...
+                ),
+                SnQLFunction(
+                    "latest_event",
+                    snql_aggregate=lambda _, alias: Function(
+                        "argMax",
+                        [self.column("id"), self.column("timestamp")],
+                        alias,
+                    ),
+                    default_result_type="integer",
```
Comment (shruthilayaj) [inline src/sentry/search/events/fields.py:2206]:
```suggestion
                    default_result_type="string",
```

Comment (shruthilayaj) [reply src/sentry/search/events/fields.py:2206]: This SnQLFunction class subclasses the DiscoverFunction class. So `snql_aggregate` replaces one of these: `aggregate` or `transform` or `conditional_transform` or `column` on the functions that appear in the original [FUNCTIONS](https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1525) dictionary. We'll still want to copy over all the other args like `default_result_type`, `redundant_grouping` and their values that appear in the [original version](https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1591-L1592)

File: src/sentry/search/events/fields.py [hunk 3]
```diff
@@ -2178,6 +2178,42 @@ def __init__(self, dataset: Dataset, params: ParamsType):
...
+                ),
+                SnQLFunction(
+                    "failure_rate",
+                    snql_aggregate=lambda _, alias: Function(
+                        "failure_rate",
+                        [],
+                        alias,
+                    ),
+                    default_result_type="integer",
```
Comment (shruthilayaj) [inline src/sentry/search/events/fields.py:2215]:
```suggestion
                    default_result_type="percentage",
```

**Comment by edwardgou-sentry** [comment]
@shruthilayaj oh good point, didnt notice this! Committed the fix!

**Review by shruthilayaj — APPROVED** [review]
LGTM

**Pull request merged**